libutils/glob_lib.h

```diff
@@ -0,0 +1,97 @@
+/*
+  Copyright 2023 Northern.tech AS
+
+  This file is part of CFEngine 3 - written and maintained by Northern.tech AS.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  To the extent this program is licensed as part of the Enterprise
+  versions of CFEngine, the applicable Commercial Open Source License
+  (COSL) may apply to this file if you as a licensee so wish it. See
+  included file COSL.txt.
+*/
+
+#ifndef GLOB_LIB_H
+#define GLOB_LIB_H
+
+#include <sequence.h>
+#include <set.h>
+
+#ifdef WITH_PCRE2
```

Comment:
For this you'd need to `#include <platform.h>`  (or `config.h`), but we can't do that in `.h` files. Please see [`regex.h.in`](https://github.com/NorthernTechHQ/libntech/blob/master/libutils/regex.h.in#L30) for how this needs to be done.